pkg/messaging/mqtt/pubsub.go

```diff
@@ -207,7 +209,7 @@ func (sub subscription) indexOf(element string) int {
 }
 
 // Deletes a topic from the slice
-func (sub subscription) delete(topic string) bool {
+func (sub *subscription) delete(topic string) bool {
```

Comment:
Use `s` instead of sub. Do the same for `indexOf` and `contains` methods.

Reply:
Also, make comments consistent - make them a full sentence starting with a capital letter and ending with a period.